pkg/local_object_storage/metabase/put.go

```diff
@@ -377,18 +379,58 @@ func decodeList(data []byte) (lst [][]byte, err error) {
 	if len(data) == 0 {
 		return nil, nil
 	}
-	r := io.NewBinReaderFromBuf(data)
-	l := r.ReadVarUint()
-	lst = make([][]byte, l, l+1)
-	for i := range lst {
-		lst[i] = r.ReadVarBytes()
+
+	var offset uint64
+	size, n, err := getVarUint(data)
+	if err != nil {
+		return nil, err
 	}
-	if r.Err != nil {
-		return nil, r.Err
+
+	offset += uint64(n)
+	lst = make([][]byte, size, size+1)
+	for i := range lst {
+		sz, n, err := getVarUint(data[offset:])
+		if err != nil {
+			return nil, err
+		}
+		offset += uint64(n)
+
+		next := offset + sz
+		if uint64(len(data)) < next {
+			return nil, gio.ErrUnexpectedEOF
+		}
+		lst[i] = data[offset:next]
+		offset = next
```

Comment:
Is  optimized version of `ReadVarBytes`? Fully compatible but has less allocation because of some use-case assumptions in the code?

Reply:
Yes, the on-disk format is the same, will add a test.